tests/integration/targets/alternatives/tasks/slaves.yml

```
@@ -0,0 +1,35 @@
+- name: Try with slaves
+  alternatives:
+    name: dummymain
+    path: '/usr/bin/dummy1'
+    link: '/usr/bin/dummymain'
+    slaves:
+      - name: dummyslave
+        path: '/usr/bin/dummy2'
+        link: '/usr/bin/dummyslave'
+  ignore_errors: True
+  register: alternative
+
+- name: check expected command was executed
+  assert:
+    that:
+      - 'alternative is successful'
+      - 'alternative is changed'
+
+- name: Execute the current dummymain command
+  shell: dummymain
```

The `shell` module should only be used if you actually need the shell to interpret things. The `command` module is sufficient to run a command:
```suggestion
  command: dummymain
```
(Same below.)